src/ca/legacy/pcas/generic/casDGClient.cc

```diff
@@ -332,7 +332,12 @@ caStatus casDGClient::searchResponse ( const caHdrLargeArray & msg,
     //
     if ( status == S_cas_success ) {
         AlignedWireRef < epicsUInt16 > tmp ( *pMinorVersion );
-        tmp = CA_MINOR_PROTOCOL_REVISION;
+        if (retVal.getMinorProtocol()) {
+          tmp = retVal.getMinorProtocol();
+        }
+        else {
+          tmp = CA_MINOR_PROTOCOL_REVISION;
+        }
```

Comment:
The value returned bt `getMinorProtocol()` needs to be bounds checked to the range `[1, CA_MINOR_PROTOCOL_REVISION]` inclusive.

Also, it would be nice to avoid calling `getMinorProtocol()` twice.

Reply:
I will fix calling twice and check for values <1. I'm not sure I agree about capping the value at CA_MINOR_PROTOCOL_REVISION. The whole point of this change is to allow a PCAS (like a name server) to return a different server's version. The specific problem being addressed is when the "real" server has a version lower than the nameserver, but in the future the real server could also have a higher version than the PCAS.